test/math/SafeMath.test.js

```diff
@@ -10,134 +10,257 @@ contract('SafeMath', function () {
-        const a = new BigNumber(284);
+  describe('signed', function () {
+    describe('add', function () {
+      it('adds correctly if it does not overflow and the result is positve', async function () {
+        const a = new BigNumber(1234);
         const b = new BigNumber(5678);
 
-        (await this.safeMath.mod(a, b)).should.be.bignumber.equal(a.mod(b));
+        (await this.safeMath.addUints(a, b)).should.be.bignumber.equal(a.plus(b));
+      });
+
+      it('adds correctly if it does not overflow and the result is negative', async function () {
+        const a = MAX_INT256;
+        const b = MIN_INT256;
+
+        const result = await this.safeMath.addInts(a, b);
+        result.should.be.bignumber.equal(a.plus(b));
+      });
+
+      it('throws a revert error on positive addition overflow', async function () {
+        const a = MAX_INT256;
+        const b = new BigNumber(1);
+
+        await shouldFail.reverting(this.safeMath.addInts(a, b));
       });
 
-      it('when the dividend is equal to the divisor', async function () {
+      it('throws a revert error on negative addition overflow', async function () {
+        const a = MIN_INT256;
+        const b = new BigNumber(-1);
+
+        await shouldFail.reverting(this.safeMath.addInts(a, b));
+      });
+    });
+
+    describe('sub', function () {
+      it('subtracts correctly if it does not overflow and the result is positive', async function () {
         const a = new BigNumber(5678);
-        const b = new BigNumber(5678);
+        const b = new BigNumber(1234);
 
-        (await this.safeMath.mod(a, b)).should.be.bignumber.equal(a.mod(b));
+        const result = await this.safeMath.subInts(a, b);
+        result.should.be.bignumber.equal(a.minus(b));
       });
 
-      it('when the dividend is larger than the divisor', async function () {
-        const a = new BigNumber(7000);
+      it('subtracts correctly if it does not overflow and the result is negative', async function () {
+        const a = new BigNumber(1234);
         const b = new BigNumber(5678);
 
-        (await this.safeMath.mod(a, b)).should.be.bignumber.equal(a.mod(b));
+        const result = await this.safeMath.subInts(a, b);
+        result.should.be.bignumber.equal(a.minus(b));
+      });
+
+      it('throws a revert error on positive subtraction overflow', async function () {
+        const a = MAX_INT256;
+        const b = new BigNumber(-1);
+
+        await shouldFail.reverting(this.safeMath.subInts(a, b));
+      });
+
+      it('throws a revert error on negative subtraction overflow', async function () {
+        const a = MIN_INT256;
+        const b = new BigNumber(1);
+
+        await shouldFail.reverting(this.safeMath.subInts(a, b));
       });
+    });
 
-      it('when the dividend is a multiple of the divisor', async function () {
-        const a = new BigNumber(17034); // 17034 == 5678 * 3
+    describe('mul', function () {
```

Comment:
The special case `(a != -1 || b != INT256_MIN)` is not being tested here (this will be made evident when the `require`s are split).